src/tests/ftest/control/dmg_config_generate.py

```diff
@@ -0,0 +1,327 @@
+    once the generated configuration is propagated to the servers.
+
+    This test assumes that systems have both nvme and dcpm available.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a ConfigGenerate object."""
+        super(ConfigGenerate, self).__init__(*args, **kwargs)
+        self._start_servers = False
+
+    def start_server_managers(self):
+        """Start the daos_server processes on each specified list of hosts."""
+        if self._start_servers:
+            super(ConfigGenerate, self).start_server_managers()
+
+    def get_storage_info(self):
+        """Get storage and network info expected from generated config file.
+
+        This function uses dmg storage scan to extract the data we need to
+        verify the generate config file.
+
+        The interface value is obtained from the OFI_INTERFACE env variable,
+        the number of NUMA nodes available is obtained from the sockets of pmem
+        devices and nvme information is matched with pmem sockets.
+
+        """
+        host = self.hostlist_servers[0]
+        data = self.get_dmg_command().storage_scan(verbose=True)
+        pmem_info = data[host]["scm"]
+        nvme_info = data[host]["nvme"]
+
+        self.log.debug("## pmem_info = {}".format(pmem_info))
+        self.log.debug("## nvme_info = {}".format(nvme_info))
+
+        interface = os.environ.get("OFI_INTERFACE")
+        pmem_sockets = set(pmem_info[p]["socket"] for p in pmem_info)
+
+        self.log.debug("## interface = {}".format(interface))
+        self.log.debug("## pmem_sockets = {}".format(pmem_sockets))
+
+        # Get nvmes that match with pmem socket.
+        # The dict would look like so: {'0': ['0000:5e:00.0', '0000:5f:00.0']}
+        sockets_info = {socket: [] for socket in pmem_sockets}
+        for nvme in nvme_info:
+            if nvme_info[nvme]["socket"] in sockets_info:
+                sockets_info[nvme_info[nvme]["socket"]].append(nvme)
+        
+        self.log.debug("## sockets_info = {}".format(sockets_info))
+        self.log.debug("## sockets_info values = {}".format(sockets_info.values()))
+
+        return {
+            "pmem": len(sockets_info),
+            "nvme": len(list(sockets_info.values())[0]),
+            "net": interface
+        }
+
+    def verify_config(self, pmem, nvme, net):
+        """Verify expected number of ioservers/nvme devices were started.
+
+        Args:
+            pmem (str): Number of SCM devices required per storage host
+            nvme (str): Minimum number of NVMe devices required per storage host
+            net (str): Network class preferred
+        """
+        # Get the generated values and the system storage info
+        sys_info = self.get_storage_info()
+        #gen_info = self.server_managers[-1].manager.job.discovered_yaml
+        gen_info = self.server_managers[-1].generated_yaml
+        self.log.debug("## gen_info = {}".format(gen_info))
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions